packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts

```diff
@@ -0,0 +1,155 @@
+import { hasFilter } from "@calcom/features/filters/lib/hasFilter";
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import { markdownToSafeHTML } from "@calcom/lib/markdownToSafeHTML";
+import { EventTypeRepository } from "@calcom/lib/server/repository/eventType";
+import { UserRepository } from "@calcom/lib/server/repository/user";
+import type { PrismaClient } from "@calcom/prisma";
+import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TGetEventTypesFromGroupSchema } from "./getByViewer.schema";
+
+type GetByViewerOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: TGetEventTypesFromGroupSchema;
+};
+
+type EventType = Awaited<ReturnType<typeof EventTypeRepository.findAllByUpId>>[number];
+
+export const getEventTypesFromGroup = async ({ ctx, input }: GetByViewerOptions) => {
+  await checkRateLimitAndThrowError({
+    identifier: `eventTypes:getEventTypesFromGroup:${ctx.user.id}`,
+    rateLimitingType: "common",
+  });
+
+  const userProfile = ctx.user.profile;
+  const { group, limit, cursor, filters } = input;
+  const { teamId, parentId } = group;
+
+  const isFilterSet = (filters && hasFilter(filters)) || !!teamId;
+  const isUpIdInFilter = filters?.upIds?.includes(userProfile.upId);
+
+  const shouldListUserEvents =
+    !isFilterSet || isUpIdInFilter || (isFilterSet && filters?.upIds && !isUpIdInFilter);
+
+  const eventTypes: EventType[] = [];
+
+  if (shouldListUserEvents || !teamId) {
+    const userEventTypes =
+      (await EventTypeRepository.findAllByUpId(
+        {
+          upId: userProfile.upId,
+          userId: ctx.user.id,
+        },
+        {
+          where: {
+            teamId: null,
+            // TODO: FIX THIS
+            // schedulingType: { not: SchedulingType.MANAGED },
+            // schedulingType: { in: [SchedulingType.ROUND_ROBIN, SchedulingType.COLLECTIVE, null] },
+            ...(isFilterSet && !!filters?.schedulingTypes
```

Comment:
```suggestion
            ...(!!filters?.schedulingTypes
```

should be sufficient